lib: set default ICU locale per cli option `--icu-locale` (#50538)
nodejs/node: doc/api/cli.md

This flag is inherited from V8 and is subject to change upstream. It may
disappear in a non-semver-major release.

### `--locale`

<!-- YAML
added: REPLACEME
-->

Set the default locale used by ICU (`Intl` object). It expects a string
representing the language version as defined in [RFC 5646][] (also known as

> BCP 47). (Overrides `LOCALE`.)

Reviewer: ```suggestion
BCP 47). (Overrides `NODE_OPTIONS`.)
```

Print node command-line options.
The output of this option is less detailed than this document.

### `--icu-locale`

<!-- YAML
added: REPLACEME
-->

Set the default locale used by ICU (`Intl` object). It expects a string

> representing the language version as defined in [RFC 5646][] (also known as
> BCP 47).

Reviewer 2: ```suggestion
representing the language version as defined in [IETF BCP 47][].
```

don't refer to 5646, that's the reason there's a BCP 47.

[OSSL_PROVIDER-legacy]: https://www.openssl.org/docs/man3.0/man7/OSSL_PROVIDER-legacy.html
[Permission Model]: permissions.md#permission-model
[REPL]: repl.md

> [RFC 5646]: https://www.rfc-editor.org/rfc/rfc5646.txt

Reviewer 2: ```suggestion
[BCP 47]: https://www.rfc-editor.org/info/bcp47
```
